cds_ils/migrator/api.py

```diff
@@ -250,15 +264,232 @@ def import_items_from_json(dump_file, include, rectype="item"):
                     continue
                 try:
                     clean_item_record(record)
-                except ItemMigrationError:
+                except ItemMigrationError as e:
+                    click.secho(str(e), fg="red")
                     continue
-                record = import_record(
-                    record, model, provider, legacy_id_key="barcode"
+                try:
+                    # check if the item already there
+                    item = get_item_by_barcode(record["barcode"])
```

Comment:
Since the legacy barcode should be unique, if you prefer you can mint an extra PID for each item with the barcode so the search is a primary key lookup instead of ES.

Reply:
I'm not 100% sure if all the barcodes are unique. but it would speed up the process for sure.